### PR TITLE
[Bug][Connector-V2][ES]Fix es source no data

### PR DESCRIPTION
**Pull request by kpretty**
## Purpose of this pull request
repair the index of complex mapping, the data cannot be obtained through the form of a.b
For example
index of mapping
```json
{
  "arkime_sessions3-221123": {
    "mappings": {
      "properties": {
        "@timestamp": {
          "type": "date"
        },
        "@version": {
          "type": "text",
          "fields": {
            "keyword": {
              "type": "keyword",
              "ignore_above": 256
            }
          }
        },
        "agent": {
          "properties": {
            "build": {
              "properties": {
                "original": {
                  "type": "keyword",
                  "ignore_above": 1024
                }
              }
            },
            "ephemeral_id": {
              "type": "keyword",
              "ignore_above": 1024
            },
            "id": {
              "type": "keyword",
              "ignore_above": 1024
            },
            "name": {
              "type": "keyword",
              "ignore_above": 1024
            },
            "type": {
              "type": "keyword",
              "ignore_above": 1024
            },
            "version": {
              "type": "keyword",
              "ignore_above": 1024
            }
          }
        }
      }
    }
  }
}
```
The current code configuration is as follows: no data can be obtained
```conf
source = ["agent.name"]
```
This PR fixes this bug
## Check list

* [x] Code changed are covered with tests, or it does not need tests for reason:
* [x] If any new Jar binary package adding in your PR, please add License Notice according
  [New License Guide](https://github.com/apache/incubator-seatunnel/blob/dev/docs/en/contribution/new-license.md)
* [x] If necessary, please update the documentation to describe the new feature. https://github.com/apache/incubator-seatunnel/tree/dev/docs
* [x] If you are contributing the connector code, please check that the following files are updated:
  1. Update change log that in connector document. For more details you can refer to [connector-v2](https://github.com/apache/incubator-seatunnel/tree/dev/docs/en/connector-v2)
  2. Update [plugin-mapping.properties](https://github.com/apache/incubator-seatunnel/blob/dev/plugin-mapping.properties) and add new connector information in it
  3. Update the pom file of [seatunnel-dist](https://github.com/apache/incubator-seatunnel/blob/dev/seatunnel-dist/pom.xml)
* [x] Update the [`release-note`](https://github.com/apache/incubator-seatunnel/blob/dev/release-note.md).

### CODE REVIEW

**Review by EricJoy2048 — DISMISSED** [review]

File: seatunnel-connectors-v2/connector-elasticsearch/src/main/java/org/apache/seatunnel/connectors/seatunnel/elasticsearch/client/EsRestClient.java [hunk 1]
```diff
@@ -376,8 +376,9 @@ public Map<String, String> getFieldTypeMapping(String index, List<String> source
                     mapping = getFieldTypeMappingFromProperties(properties, source);
                 } else {
                     for (Iterator<JsonNode> iter = mappingsProperty.iterator(); iter.hasNext(); ) {
-                        JsonNode typeNode = iter.next();
-                        JsonNode properties = typeNode.get("properties");
+                        // JsonNode typeNode = iter.next();
+                        // JsonNode properties = typeNode.get("properties");
```
Comment (hailin0) [inline seatunnel-connectors-v2/connector-elasticsearch/src/main/java/org/apache/seatunnel/connectors/seatunnel/elasticsearch/client/EsRestClient.java]: remove

Comment (kpretty) [reply seatunnel-connectors-v2/connector-elasticsearch/src/main/java/org/apache/seatunnel/connectors/seatunnel/elasticsearch/client/EsRestClient.java]: useless code removed

**Review by hailin0 — DISMISSED** [review]

File: seatunnel-connectors-v2/connector-elasticsearch/src/main/java/org/apache/seatunnel/connectors/seatunnel/elasticsearch/client/EsRestClient.java [hunk 2]
```diff
@@ -376,8 +376,7 @@ public Map<String, String> getFieldTypeMapping(String index, List<String> source
                     mapping = getFieldTypeMappingFromProperties(properties, source);
                 } else {
                     for (Iterator<JsonNode> iter = mappingsProperty.iterator(); iter.hasNext(); ) {
-                        JsonNode typeNode = iter.next();
-                        JsonNode properties = typeNode.get("properties");
+                        JsonNode properties = iter.next();
```
Comment (TyrantLucifer) [inline seatunnel-connectors-v2/connector-elasticsearch/src/main/java/org/apache/seatunnel/connectors/seatunnel/elasticsearch/client/EsRestClient.java]: It maybe has some problems on es5 cc @iture123

**Review by TyrantLucifer — CHANGES_REQUESTED** [review]

File: seatunnel-connectors-v2/connector-elasticsearch/src/main/java/org/apache/seatunnel/connectors/seatunnel/elasticsearch/client/EsRestClient.java (hunk 2 above)
Comment (kpretty) [reply seatunnel-connectors-v2/connector-elasticsearch/src/main/java/org/apache/seatunnel/connectors/seatunnel/elasticsearch/client/EsRestClient.java]: @TyrantLucifer Indeed, the mapping of es has changed, this problem has been fixed and the test has passed es5, es6, es7+
https://github.com/apache/incubator-seatunnel/blob/ac50b31ca3a534649cb90b467a7654f6e52f7180/seatunnel-connectors-v2/connector-elasticsearch/src/main/java/org/apache/seatunnel/connectors/seatunnel/elasticsearch/client/EsRestClient.java#L378-L387

File: seatunnel-connectors-v2/connector-elasticsearch/pom.xml [hunk 3]
```diff
@@ -53,6 +53,18 @@
...
+                    <groupId>com.google.guava</groupId>
+                    <artifactId>guava</artifactId>
+                </exclusion>
+            </exclusions>
+        </dependency>
+        <!-- I'm not sure which version of guava to use, so I choose the latest; sure enough, the version inherited from the parent project is not compatible -->
+        <dependency>
+            <groupId>com.google.guava</groupId>
+            <artifactId>guava</artifactId>
+            <version>31.1-jre</version>
```
Comment (TyrantLucifer) [inline seatunnel-connectors-v2/connector-elasticsearch/pom.xml]: use properties instead of it.

Comment (kpretty) [reply seatunnel-connectors-v2/connector-elasticsearch/pom.xml]: completed

Comment (TyrantLucifer) [reply seatunnel-connectors-v2/connector-elasticsearch/pom.xml]: Thank you for your contribution. Let's waiting CICD

**Review by TyrantLucifer — APPROVED** [review]
LGTM

**Review by hailin0 — APPROVED** [review]

**Pull request merged**